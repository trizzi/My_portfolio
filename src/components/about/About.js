import React from 'react';
import Button from '../button/Button';
import SlideIn from '../../utilities/SlideIn';
import showcaseImg from '../../assets/Profile picture (1).svg';
import aboutLogo from '../../assets/about_button.svg';
import text from '../../assets/PATRICK AKHAMIOGU.svg';
import aboutImage from '../../assets/about_background_image.svg';

const About = () => {
	return (
		<div
			name='section2'
			className='md:h-[90vh] lg:h-[83vh]'
			style={{
				backgroundImage: `url(${aboutImage})`,
				backgroundRepeat: 'no-repeat',
				backgroundSize: 'cover',
				// height: '800px',
				backgroundColor: 'rgba(0,0,0,0.9)',
			}}>
			<div className='flex flex-col sm:flex-row w-full justify-between text-white'>
				<div className='bg-[#29292E] opacity-80 md:w-1/2'>
					<div className='mt-6'>
						<div className='flex px-12 pt-8'>
							<div>
								<img
									src={aboutLogo}
									alt='about-logo'
								/>
							</div>
							<div>
								<h1>About</h1>
								<img
									src={text}
									alt='Patrick-Akhamiogu'
								/>
							</div>
						</div>
						<div className='lg:ml-24 md:mx-16 mx-14'>
							<h1 className='font-bold pb-4'>Professional Background</h1>
							<p>
								With years of professional experience, I have honed my skills as
								a web developer, <br />
								focusing primarily on front-end development, complemented by
								expertise in backend <br />
								development
							</p>
						</div>
						<div className='lg:ml-24 md:mx-16 mx-14'>
							<h1 className='font-bold pb-4'>Passion and Approach</h1>
							<p>
								As a deadline-focused professional, I am recognized for my
								proactive approach, effective communication, and results-driven
								orientation. My primary interest lies in addressing tangible
								customer challenges with robust business justifications.
								Currently seeking opportunities to broaden my skills in web
								development and project management, my proficiency in Graphic
								design is integral to crafting visually stunning websites with
								exceptional UI/UX designs. Furthermore, I have a personal
								passion for music, music production, and video editing.
							</p>
						</div>
						<div className='lg:ml-24 md:mx-16 my-4 ml-14 mb-10'>
							<Button
								text={'Hire Me'}
								className='text-white'
							/>
						</div>
					</div>
				</div>
				<div
					className='hidden md:block md:visible sm:w-1/2 md:h-[920px] lg:h-[800px]'
					style={{
						backgroundImage: `url(${aboutImage})`,
						backgroundRepeat: 'no-repeat',
						backgroundSize: 'cover',
						// height: '920px',
						backgroundColor: 'rgba(0,0,0,0.9)',
					}}>
					<SlideIn>
						<div className='md:mt-[40rem] md:ml-[25rem]'>
							<div className='md:mx-10 md:my-20 '>
								<div className='hidden md:flex text-white bg-[#29292E] opacity-90 rounded-full p-4 w-fit hover:bg-black'>
									<img
										src={showcaseImg}
										alt='Images'
									/>
									<a
										href='info@ideas2launch.com'
										className='pt-3 pl-2'>
										<p>patrickakhamiogu@gmail.com</p>
									</a>
								</div>
							</div>
						</div>
					</SlideIn>
				</div>
			</div>
		</div>
	);
};

export default About;
