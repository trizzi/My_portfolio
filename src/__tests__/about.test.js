import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeContextProvider } from '../context/ThemeContext';
import About from '../components/about/About';

// Mock the IntersectionObserver
beforeEach(() => {
	window.IntersectionObserver = jest.fn(function () {
		this.observe = jest.fn();
		this.disconnect = jest.fn();
		this.unobserve = jest.fn();
	});
});

describe('About component', () => {
	test('renders About section with correct content', () => {
		const { container } = render(
			<ThemeContextProvider>
				<About />
			</ThemeContextProvider>,
		);

		const aboutSection = screen.getByTestId('about-section');
		expect(aboutSection).toBeInTheDocument();

		const aboutLogo = screen.getByAltText('about-logo');
		expect(aboutLogo).toBeInTheDocument();

		const aboutText = screen.getByAltText('Patrick-Akhamiogu');
		expect(aboutText).toBeInTheDocument();

		const professionalBackground = screen.getByText('Professional Background');
		expect(professionalBackground).toBeInTheDocument();

		const passionAndApproach = screen.getByText('Passion and Approach');
		expect(passionAndApproach).toBeInTheDocument();

		const hireMeButton = screen.getByText('Hire Me');
		expect(hireMeButton).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	test('triggers email link on email click', () => {
		const { container } = render(
			<ThemeContextProvider>
				<About />
			</ThemeContextProvider>,
		);

		const emailLink = screen.getByText('patrickakhamiogu@gmail.com');
		fireEvent.click(emailLink);

		expect(container).toMatchSnapshot();
	});
});
