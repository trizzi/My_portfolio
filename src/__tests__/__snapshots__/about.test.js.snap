// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders About section with correct content 1`] = `
<div>
  <div
    class="md:h-[90vh] lg:h-[83vh]"
    data-testid="about-section"
    name="section2"
    style="background-image: url(about_background_image.svg); background-repeat: no-repeat; background-size: cover; background-color: rgba(0, 0, 0, 0.9);"
  >
    <div
      class="flex flex-col sm:flex-row w-full justify-between text-white"
    >
      <div
        class="bg-[#29292E] opacity-80 md:w-1/2"
      >
        <div
          class="mt-6"
        >
          <div
            class="flex pl-6 pr-12 md:px-12 pt-8"
          >
            <div>
              <img
                alt="about-logo"
                src="about_button.svg"
              />
            </div>
            <div>
              <h1>
                About
              </h1>
              <img
                alt="Patrick-Akhamiogu"
                src="PATRICK AKHAMIOGU.svg"
              />
            </div>
          </div>
          <div
            class="lg:ml-24 md:mx-16 mx-14"
          >
            <h1
              class="font-bold pb-4"
            >
              Professional Background
            </h1>
            <p>
              With years of professional experience, I have honed my skills as a web developer, 
              <br />
              focusing primarily on front-end development, complemented by expertise in backend 
              <br />
              and a bit of mobile development.
            </p>
          </div>
          <div
            class="lg:ml-24 md:mx-16 mx-14"
          >
            <h1
              class="font-bold py-4"
            >
              Passion and Approach
            </h1>
            <p>
              As a deadline-focused professional, I am recognized for my proactive approach, effective communication, and results-driven orientation. My primary interest lies in addressing tangible customer challenges with robust business justifications. Currently seeking opportunities to broaden my skills in web development and project management, my proficiency in Graphic design is integral to crafting visually stunning websites with exceptional UI/UX designs. Furthermore, I have a personal passion for music, music production, and video editing.
            </p>
          </div>
          <div
            class="lg:ml-24 md:mx-16 my-4 ml-14 mb-10"
          >
            <a>
              <button
                class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
                style="background-color: transparent; color: rgb(0, 0, 0); border: 2px solid #9932cc; opacity: 0;"
                tabindex="0"
              >
                Hire Me
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="hidden md:block md:visible sm:w-1/2 md:h-[920px] lg:h-[800px]"
        style="background-image: url(about_background_image.svg); background-repeat: no-repeat; background-size: cover; background-color: rgba(0, 0, 0, 0.9);"
      >
        <div
          style="transform: translateX(-1000px) translateZ(0);"
        >
          <div
            class="md:mt-[40rem] md:ml-[25rem]"
          >
            <div
              class="md:mx-10 md:my-20 "
            >
              <div
                class="hidden md:flex text-white bg-[#29292E] opacity-90 rounded-full p-4 w-fit hover:bg-black"
              >
                <img
                  alt="Images"
                  src="profile-picture.svg"
                />
                <a
                  class="pt-3 pl-2"
                  href="mailto:patrickakhamiogu@gmail.com"
                >
                  <p>
                    patrickakhamiogu@gmail.com
                  </p>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`About component triggers email link on email click 1`] = `
<div>
  <div
    class="md:h-[90vh] lg:h-[83vh]"
    data-testid="about-section"
    name="section2"
    style="background-image: url(about_background_image.svg); background-repeat: no-repeat; background-size: cover; background-color: rgba(0, 0, 0, 0.9);"
  >
    <div
      class="flex flex-col sm:flex-row w-full justify-between text-white"
    >
      <div
        class="bg-[#29292E] opacity-80 md:w-1/2"
      >
        <div
          class="mt-6"
        >
          <div
            class="flex pl-6 pr-12 md:px-12 pt-8"
          >
            <div>
              <img
                alt="about-logo"
                src="about_button.svg"
              />
            </div>
            <div>
              <h1>
                About
              </h1>
              <img
                alt="Patrick-Akhamiogu"
                src="PATRICK AKHAMIOGU.svg"
              />
            </div>
          </div>
          <div
            class="lg:ml-24 md:mx-16 mx-14"
          >
            <h1
              class="font-bold pb-4"
            >
              Professional Background
            </h1>
            <p>
              With years of professional experience, I have honed my skills as a web developer, 
              <br />
              focusing primarily on front-end development, complemented by expertise in backend 
              <br />
              and a bit of mobile development.
            </p>
          </div>
          <div
            class="lg:ml-24 md:mx-16 mx-14"
          >
            <h1
              class="font-bold py-4"
            >
              Passion and Approach
            </h1>
            <p>
              As a deadline-focused professional, I am recognized for my proactive approach, effective communication, and results-driven orientation. My primary interest lies in addressing tangible customer challenges with robust business justifications. Currently seeking opportunities to broaden my skills in web development and project management, my proficiency in Graphic design is integral to crafting visually stunning websites with exceptional UI/UX designs. Furthermore, I have a personal passion for music, music production, and video editing.
            </p>
          </div>
          <div
            class="lg:ml-24 md:mx-16 my-4 ml-14 mb-10"
          >
            <a>
              <button
                class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
                style="background-color: transparent; color: rgb(0, 0, 0); border: 2px solid #9932cc; opacity: 0;"
                tabindex="0"
              >
                Hire Me
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="hidden md:block md:visible sm:w-1/2 md:h-[920px] lg:h-[800px]"
        style="background-image: url(about_background_image.svg); background-repeat: no-repeat; background-size: cover; background-color: rgba(0, 0, 0, 0.9);"
      >
        <div
          style="transform: translateX(-1000px) translateZ(0);"
        >
          <div
            class="md:mt-[40rem] md:ml-[25rem]"
          >
            <div
              class="md:mx-10 md:my-20 "
            >
              <div
                class="hidden md:flex text-white bg-[#29292E] opacity-90 rounded-full p-4 w-fit hover:bg-black"
              >
                <img
                  alt="Images"
                  src="profile-picture.svg"
                />
                <a
                  class="pt-3 pl-2"
                  href="mailto:patrickakhamiogu@gmail.com"
                >
                  <p>
                    patrickakhamiogu@gmail.com
                  </p>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
